File: BackEnd/internal/domain/category/category.go
package category

import "time"

type Category struct {
	CreatedAt time.Time `json:"created_at"`
	Id        string    `json:"id"`
	Name      string    `json:"name"`
	Icon      string    `json:"icon"`
	Color     string    `json:"color"`
	UserId    string    `json:"user_id"`
}

func NewCategory(id, name, icon, color string) *Category {
	return &Category{
		Id:    id,
		Name:  name,
		Icon:  icon,
		Color: color,
	}
}

File: ios-app/gestorpresupuesto/gestorpresupuesto/Domain/Entities/Category.swift
import Foundation

struct Category: Codable, Identifiable, Equatable, Hashable {
    let id: String
    let name: String
    let icon: String
    let color: String
    let userId: String?
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case icon
        case color
        case userId = "user_id"
        case createdAt = "created_at"
    }
}

struct CreateCategoryRequest: Codable {
    let name: String
    let icon: String
    let color: String
}

struct UpdateCategoryRequest: Codable {
    let name: String
    let icon: String
    let color: String
}
